PHPDaemon/Clients/DNS/Connection.php

```diff
@@ -178,6 +243,7 @@ public function onRead()
             $this->state = self::STATE_ROOT;
             $this->setWatermark(2);
             $this->onUdpPacket($pct);
+            return;
```

Comment:
Packets can be pipelined. onRead() is not going to be get triggered until new data comes in. This `return;` might leave the buffer non-empty. Please remove `return;` :)